### PR TITLE
Update pre commit and other changes

### PR DESCRIPTION
**Pull request by SanketDG**

### CODE REVIEW

File: src/bandersnatch/main.py [hunk 1]
```diff
@@ -102,7 +102,7 @@ async def async_main(args: argparse.Namespace, config: ConfigParser) -> int:
         if status_file.exists():
             tmp_status_file = Path(gettempdir()) / "status"
             try:
-                shutil.move(status_file, tmp_status_file)
+                shutil.move(str(status_file), tmp_status_file)
```
Comment (cooperlees) [inline src/bandersnatch/main.py:105]: Hmm - I thought this could handle a path-like-object but both 3.6 and 3.7 documentation don't say it can. So thanks!

**Comment by cooperlees** [comment]
P.s. codecov really confuses me. I'm thinking of dropping it :|

File: src/bandersnatch/main.py (hunk 1 above)
Comment (SanketDG) [reply src/bandersnatch/main.py:105]: Relevant (it's interesting): https://github.com/python/typeshed/pull/3559

**Comment by SanketDG** [comment]
> Awesome - Hope monkeytype is working. I use to sit 2 rows away from the guy who wrote it at Instagram :)

Oh, that's nice!

> I did plan to fire it up with running tox + the integration test to get some typing for free.

Planning to do soon.

> P.s. codecov really confuses me. I'm thinking of dropping it :|

Yeah, I was also not sure what's going on. But that's fine, every piece of software fails occasionally, some more, some less :D

**Pull request merged**